http-clients/src/test/java/org/triplea/http/client/web/socket/WebSocketConnectionTest.java

```diff
@@ -81,10 +82,19 @@ void close() {
     }
 
     @Test
-    void sendMessageWaitsForConnection() {
+    void sendMessage() {
       webSocketConnection.sendMessage(MESSAGE);
 
       verify(webSocketClient).send(MESSAGE);
     }
+
+    @Test
+    void sendMessageFailsIfConnectionNotOpened() {
+      when(webSocketClient.isOpen()).thenReturn(false);
+
+      assertThrows(IllegalStateException.class, () -> webSocketConnection.sendMessage(MESSAGE));
+
+      verify(webSocketClient, times(0)).send(MESSAGE);
```

Comment:
nit: `Mockito.never()` AFAIK is preferred over `times(0)`  (in the same way as `isEmpty()` is preferred over `length == 0`)

Reply:
Good call, I knew there was something to achieve this, but I couldn't remember that's what it was called